clients/drcachesim/tests/burst_traceopts.cpp

```diff
@@ -209,9 +209,53 @@ gather_trace(const std::string &tracer_ops, const std::string &out_subdir)
     return post_process(out_subdir);
 }
 
+static void
+reg_id_set_unit_tests()
+{
+    reg_id_set_t set;
+    assert(set.begin() == set.end());
+    // Test non-GPR.
+    auto insert_res = set.insert(IF_X86_ELSE(DR_REG_XMM0, DR_REG_Q0));
+    assert(insert_res.first == set.end());
+    assert(!insert_res.second);
+    assert(set.begin() == set.end());
+    // Test adding a GPR.
+    insert_res = set.insert(DR_REG_START_GPR + 1);
+    assert(insert_res.first != set.end());
+    assert(insert_res.second);
+    assert(set.begin() != set.end());
+    // Test find not-present.
+    auto find_res = set.find(DR_REG_START_GPR);
+    assert(find_res == set.end());
+    // Test find present.
+    find_res = set.find(DR_REG_START_GPR + 1);
+    assert(find_res != set.end());
+    assert(*find_res == DR_REG_START_GPR + 1);
+    // Test erase with two entries.
+    insert_res = set.insert(DR_REG_START_GPR + 4);
+    assert(insert_res.second);
+    auto iter = set.begin();
+    bool found_next = false;
+    while (iter != set.end()) {
+        if (*iter == DR_REG_START_GPR + 1)
+            iter = set.erase(iter);
+        else {
+            if (*iter == DR_REG_START_GPR + 4)
+                found_next = true;
+            ++iter;
+        }
+    }
+    assert(found_next);
+    assert(set.find(DR_REG_START_GPR + 1) == set.end());
+    assert(set.find(DR_REG_START_GPR + 4) != set.end());
+    assert(*set.find(DR_REG_START_GPR + 4) == DR_REG_START_GPR + 4);
+}
```

Comment:
Perhaps add a test to show that a reg added multiple times, but removed only once, is indeed removed from the set.

Reply:
Done.